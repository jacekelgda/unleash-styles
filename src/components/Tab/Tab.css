.Tab {
  display: inline-block;
  margin-right: 2rem;
}
.TabLink {
  display: inline-block;
  font: 600 14px/70px 'Montserrat', sans-serif;
  text-transform: uppercase;
  text-decoration: none;
  color: #b8b7bc;
  border-bottom: 2px solid transparent;
  transition: color .3s, border-bottom-color .3s;
}
.TabLink:hover,
.TabLink:active {
  color: #f33c57;
  border-bottom-color: #f33c57;
}
