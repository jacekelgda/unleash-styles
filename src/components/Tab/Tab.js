import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styles from './Tab.css'

/**
 * Tab element on tabs component
 */
class Tab extends Component {
  render() {
    const { label } = this.props
    return (
      <li className={styles.Tab}>
        <a href="index.html" className={styles.TabLink}>
          <span>{label}</span>
        </a>
      </li>
    )
  }
}

Tab.propTypes = {
  /** Text on tab */
  label: PropTypes.string,
}

export default Tab
