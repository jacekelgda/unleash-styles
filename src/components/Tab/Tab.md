Tab component:

    <Tab label="Label" />
