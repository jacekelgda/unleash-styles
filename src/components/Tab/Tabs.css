.Tabs {
  line-height: 70px;
}

.TabsList {
  padding: 0;
  margin: 0;
  list-style: none;
}

.Tabs_mobile {
  display: none;
}

@media screen and (max-width: 767px) {
  .Tabs {
		display: block;
		font-size: 14px;
		text-align: center;
		border: none;
		display: block;
		position: relative;
		padding: 18px 0;
		z-index: 10;
		top: 0;
		left: 0;
		width: 100%;
		background-color: #fff;
		line-height: 36px;
	}
}
