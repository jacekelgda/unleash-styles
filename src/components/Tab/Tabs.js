import React, { Component } from 'react'
import styles from './Tabs.css'

/**
 * Tabs component that hosts tab components
 */
class Tabs extends Component {
  render() {
    return (
      <nav className={styles.Tabs}>
    		<ul className={styles.TabsList}>
          {this.props.children}
    		</ul>
    	</nav>
    )
  }
}

export default Tabs
