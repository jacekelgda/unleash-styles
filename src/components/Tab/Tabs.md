Tabs component:

    <Tabs>
      <Tab label="one" />
      <Tab label="two" />
      <Tab label="three" />
    </Tabs>
