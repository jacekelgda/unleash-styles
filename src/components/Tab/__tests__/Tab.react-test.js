/* global test, expect */
import React from 'react'
import { shallow } from 'enzyme'
import Tabs from '../Tabs'
import Tab from '../Tab'

test('Tabs renders correctly', () => {
  const wrapper = shallow(
    <Tabs>
      <Tab label='test' isCurrent />
      <Tab label='test2' />
    </Tabs>
  )
  expect(wrapper).toMatchSnapshot()
})
