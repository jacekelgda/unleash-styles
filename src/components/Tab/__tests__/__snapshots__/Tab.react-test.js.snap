// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <nav
    className="Tabs"
>
    <ul
        className="TabsList"
    >
        <Tab
            label="test"
        />
        <Tab
            label="test2"
        />
    </ul>
</nav>,
  "nodes": Array [
    <nav
      className="Tabs"
>
      <ul
            className="TabsList"
      >
            <Tab
                  label="test"
            />
            <Tab
                  label="test2"
            />
      </ul>
</nav>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Tabs>
        <Tab
                label="test"
        />
        <Tab
                label="test2"
        />
</Tabs>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Tabs {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": Array [
            <Tab
              label="test"
/>,
            <Tab
              label="test2"
/>,
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <nav
          className="Tabs"
>
          <ul
                    className="TabsList"
          >
                    <Tab
                              label="test"
                    />
                    <Tab
                              label="test2"
                    />
          </ul>
</nav>,
        "_debugID": 2,
        "_renderedOutput": <nav
          className="Tabs"
>
          <ul
                    className="TabsList"
          >
                    <Tab
                              label="test"
                    />
                    <Tab
                              label="test2"
                    />
          </ul>
</nav>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Tabs>
    <Tab
        label="test"
    />
    <Tab
        label="test2"
    />
</Tabs>,
}
`;
